assets/content/cookbook/Macros/autobuild-fields-from-enum.md

```diff
@@ -0,0 +1,143 @@
+[tags]: / "autobuild-macro,building-fields"
+
+# Autobuild fields from enum type param
+
+This macro function adds fields corresponding to an extended type's enum type param.
+
+```haxe
+import haxe.macro.Expr;
+import haxe.macro.Context;
+import haxe.macro.Type;
+
+using haxe.macro.Tools;
+
+class Macro {
+  public static function buildSet():Array<Field> {
+    final fields = Context.getBuildFields();
+    
+    try {
+      return switch Context.getLocalType() {
+        // Extract the type parameter
+        case TInst(getEnumType(_.get()) => enumType, _):
+          // Generate fields
+          addFields(enumType, fields);
+
+          fields;
+        case found:
+          throw 'Expected TInst, found: $found';
+      }
+    } catch(e) {
+      // Catch thrown errors and point them to the failing class
+      Context.error(e.message, Context.currentPos());
+      return Context.getBuildFields();
+    }
```

Comment:
I've found that this is my preferred way of handling macro errors, using throw everywhere, catching them and converting them to a `Context.error` that points to the line that triggered the macro. This is more informative to the dev utilizing the macro, but less informative to devs adding to or maintaining the macro, which I fix by temporarily commenting out the try/catch.

Would it be better or simpler to just use Context.error everywhere and remove the try? Alternatively should I trim down the error handling altogether and just have a simpler but perhaps more fragile example that just needs to work for the given usage example?